core/utils.py

```diff
@@ -565,7 +565,7 @@ def convert_to_hash(input_string: str, max_length: int) -> str:
     # Prefixing altchars with b' to ensure that all characters in encoded_string
     # remain encoded (otherwise encoded_string would be of type unicode).
     encoded_string = base64.b64encode(
-        hashlib.sha1(input_string.encode('utf-8')).digest(),
+        hashlib.sha224(input_string.encode('utf-8')).digest(),
```

Comment:
I don't think we can change this. The results of this are stored in some places (for example `DeletedUsernameModel`) and by introducing this we would have to fix all the already stored values.

Reply:
@vojtechjelinek In that case, should we ignore this file in the CodeQL config too? Can you also clarify what you mean by adding a test that detects this change? Thank you!

Reply:
I think this will need to be ignored and either ignored forever or replaced everywhere using a migration. 

I think that for our use-case sha1 is fine as we don't use it for enciphering but rather for reducing and anonymizing data. Afaik preimage attacks (getting the plaintext value from the hash) are not easy and thus there is no risk for us.

I don't think that the test is such a good idea, we can skip it.

Reply:
Oh okay I see, thanks! Ignored this file